### PR TITLE
MARKET-1303 Gauge Chart

### PR DESCRIPTION
**Pull request by Saharij**
Screenshot 2022-08-01 at 13 35 59
Screenshot 2022-08-01 at 13 33 52
Screenshot 2022-08-01 at 13 31 48

### CODE REVIEW

**Review by StasivPavlo — COMMENTED** [review]
👍

File: components/bl-gauge-chart-component/src/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export default function GaugeChartComponent({ component }) {
+  const { classList, display, disable, goal, progress } = component;
+
+  const percentOfProgress = (progress / (goal / 100)).toFixed(2);
+  const percentOfTurn = (percentOfProgress > 100 ? 100 : percentOfProgress) / 200;
+
+  const fillTurnPercentStyle = { transform: `rotate(${percentOfTurn}turn)` };
+
+  const decoratedGoal = decorateNumber(String(goal));
```
Comment (v-excelsior) [inline components/bl-gauge-chart-component/src/index.js:13]: Not critical, but I think 'shownGoal' variable name fit better.

File: components/bl-gauge-chart-component/src/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          <div className="gauge-chart__fill" style={fillTurnPercentStyle}></div>
+          <div className="gauge-chart__cover">
+            <span className="gauge-chart__progress-number">{ decoratedProgress }</span>
+            <div className="gauge-chart__progress-percent">
+              <span>{ `${ percentOfProgress }% ` }</span>
+              <span>complete</span>
+            </div>
+          </div>
+        </div>
+        <div className="gauge-chart__info">
```
Comment (v-excelsior) [inline components/bl-gauge-chart-component/src/index.js:35]: Into external component? Could be in this file but below main component.

File: components/bl-gauge-chart-component/src/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { decorateNumber } from './decorate-number';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function GaugeChartComponent({ component }) {
+  const { classList, display, disable, goal, progress } = component;
+
+  const percentOfProgress = (progress / (goal / 100)).toFixed(2);
+  const percentOfTurn = (percentOfProgress > 100 ? 100 : percentOfProgress) / 200;
```
Comment (v-excelsior) [inline components/bl-gauge-chart-component/src/index.js:9]: It's not an angle of turn?

File: components/bl-gauge-chart-component/src/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { decorateNumber } from './decorate-number';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function GaugeChartComponent({ component }) {
+  const { classList, display, disable, goal, progress } = component;
+
+  const percentOfProgress = (progress / (goal / 100)).toFixed(2);
```
Comment (v-excelsior) [inline components/bl-gauge-chart-component/src/index.js:8]: Also, how about move all this calculation into hook 'useChartData' or somethink like this.

In this case component became more light in first look.

`const { progressAngle, progressPercentage, shownProgress, shownGoal } = useChartData(goal, progress)`

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-gauge-chart-component/src/decorate-number.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+const getFirstNumbers = (value, slicer) => {
```
[the remark this answers is not in the document]
Comment (v-excelsior) [reply components/bl-gauge-chart-component/src/decorate-number.js:1]: Or even better

```Intl.NumberFormat('en', { notation: 'compact' }).format( num );```

**Pull request merged**